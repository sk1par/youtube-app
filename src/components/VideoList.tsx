import { useSelector } from 'react-redux';
import IVideos from '../models/videos.interface';
import { allVideos } from '../selectors/selectors';
import VideoItem from './VideoItem';

const VideoList = () => {
    const videos = useSelector(allVideos);


    return (
        videos.length > 0 ? <div className='ui relaxed divided list' data-auto-id="list-component" >
            {videos.map((video: IVideos) =>
                <VideoItem key={video.id.videoId} video={video} />
            )}
        </div> : null
    )
};
export default VideoList;