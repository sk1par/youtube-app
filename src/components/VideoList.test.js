import { useSelector } from 'react-redux';
import VideoList from './VideoList';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallowToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: () => (fn) => () => fn(),
  useSelector: jest.fn(() => []),
}));

jest.mock('../actions/videosActions', () => {
  const selectedVideo = jest.fn();
  return selectedVideo;
});

const getVideo = () => ({
    id: {
      videoId: '1',
    },
    snippet: {
      description: 'test',
      title: 'test',
      thumbnails: {
        medium: {
          url: 'test',
        },
      },
    },
  });

it('should render correctly', () => {
  const v = getVideo();
  useSelector.mockReturnValueOnce([v]);
  const wrapper = shallow(<VideoList />);
  const title = wrapper.find('[data-auto-id="list-component"]');
  expect(title).toHaveLength(1);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});