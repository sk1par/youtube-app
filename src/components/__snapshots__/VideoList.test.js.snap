// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="ui relaxed divided list"
  data-auto-id="list-component"
>
  <VideoItem
    key="1"
    video={
      Object {
        "id": Object {
          "videoId": "1",
        },
        "snippet": Object {
          "description": "test",
          "thumbnails": Object {
            "medium": Object {
              "url": "test",
            },
          },
          "title": "test",
        },
      }
    }
  />
</div>
`;
